src/Features/Core/Portable/CodeRefactorings/AddNamedArguments/AbstractAddNamedArgumentsCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.CodeRefactorings.AddNamedArguments
+{
+    internal abstract class AbstractAddNamedArgumentsCodeRefactoringProvider<TArgumentSyntax> : CodeRefactoringProvider where TArgumentSyntax : SyntaxNode
+    {
+        public sealed override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        {
+            var document = context.Document;
+            if (document.Project.Solution.Workspace.Kind == WorkspaceKind.MiscellaneousFiles)
+            {
+                return;
+            }
+
+            var cancellationToken = context.CancellationToken;
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+            var node = root.FindNode(context.Span).FirstAncestorOrSelf<SyntaxNode>(IsCandidate);
+            if (node?.ContainsDiagnostics != false)
```

Comment:
i find boolean questions with ?. very confusing.  Can you first check for null, and return in that case.  Then just check ContainsDiagnostics.